Unable to build Horovod without CUDA support

**Environment:**

1. Framework: Tensorflow
2. Framework version: 2.1.0
3. Horovod version: 0.19.1, master
4. MPI version: OpenMPI 3.1.5
5. CUDA version: N/A
6. NCCL version: N/A
7. Python version: 3.7.6
8. OS and version: macOS 10.15.3
9. GCC version: Clang 11.0.3

**Bug report:**

I'm trying to build Horovod on my laptop, which does not have CUDA support, but the installation fails with the following error message:
```
build/temp.macosx-10.15.3-x86_64-3.7/test_compile/test_cuda.cc:1:10: fatal error: 'cuda_runtime.h' file not found
#include <cuda_runtime.h>
         ^~~~~~~~~~~~~~~~
1 error generated.
error: CUDA library was not found (see error above).
Please specify correct CUDA location with the HOROVOD_CUDA_HOME environment variable or combination of HOROVOD_CUDA_INCLUDE and HOROVOD_CUDA_LIB environment variables.

HOROVOD_CUDA_HOME - path where CUDA include and lib directories can be found
HOROVOD_CUDA_INCLUDE - path to CUDA include directory
HOROVOD_CUDA_LIB - path to CUDA lib directory
```
Is it not possible to build Horovod without CUDA support?

Here is my build log and environment:

* [spack-build-out.txt](https://github.com/horovod/horovod/files/4394576/spack-build-out.txt)
* [spack-build-env.txt](https://github.com/horovod/horovod/files/4394577/spack-build-env.txt)

Thanks @tgaddair! That fixed my issue. I'm now experiencing a different issue while building the TF backend:
```
horovod/common/ops/adasum/adasum.h:445:23: error: always_inline function '_mm256_fmadd_pd' requires target feature 'fma', but would be inlined into function 'ComputeDotAndNormSqrdsfp16' that is compiled without support for 'fma'
      dotProductVec = _mm256_fmadd_pd(aBot, bBot, dotProductVec);
                      ^
horovod/common/ops/adasum/adasum.h:446:23: error: always_inline function '_mm256_fmadd_pd' requires target feature 'fma', but would be inlined into function 'ComputeDotAndNormSqrdsfp16' that is compiled without support for 'fma'
      dotProductVec = _mm256_fmadd_pd(aTop, bTop, dotProductVec);
                      ^
horovod/common/ops/adasum/adasum.h:447:18: error: always_inline function '_mm256_fmadd_pd' requires target feature 'fma', but would be inlined into function 'ComputeDotAndNormSqrdsfp16' that is compiled without support for 'fma'
      anormVec = _mm256_fmadd_pd(aBot, aBot, anormVec);
                 ^
horovod/common/ops/adasum/adasum.h:448:18: error: always_inline function '_mm256_fmadd_pd' requires target feature 'fma', but would be inlined into function 'ComputeDotAndNormSqrdsfp16' that is compiled without support for 'fma'
      anormVec = _mm256_fmadd_pd(aTop, aTop, anormVec);
                 ^
horovod/common/ops/adasum/adasum.h:449:18: error: always_inline function '_mm256_fmadd_pd' requires target feature 'fma', but would be inlined into function 'ComputeDotAndNormSqrdsfp16' that is compiled without support for 'fma'
      bnormVec = _mm256_fmadd_pd(bBot, bBot, bnormVec);
                 ^
horovod/common/ops/adasum/adasum.h:450:18: error: always_inline function '_mm256_fmadd_pd' requires target feature 'fma', but would be inlined into function 'ComputeDotAndNormSqrdsfp16' that is compiled without support for 'fma'
      bnormVec = _mm256_fmadd_pd(bTop, bTop, bnormVec);
                 ^
horovod/common/ops/adasum/adasum.h:459:23: error: always_inline function '_mm256_fmadd_pd' requires target feature 'fma', but would be inlined into function 'ComputeDotAndNormSqrdsfp16' that is compiled without support for 'fma'
      dotProductVec = _mm256_fmadd_pd(aBot, bBot, dotProductVec);
                      ^
horovod/common/ops/adasum/adasum.h:460:23: error: always_inline function '_mm256_fmadd_pd' requires target feature 'fma', but would be inlined into function 'ComputeDotAndNormSqrdsfp16' that is compiled without support for 'fma'
      dotProductVec = _mm256_fmadd_pd(aTop, bTop, dotProductVec);
                      ^
horovod/common/ops/adasum/adasum.h:461:18: error: always_inline function '_mm256_fmadd_pd' requires target feature 'fma', but would be inlined into function 'ComputeDotAndNormSqrdsfp16' that is compiled without support for 'fma'
      anormVec = _mm256_fmadd_pd(aBot, aBot, anormVec);
                 ^
horovod/common/ops/adasum/adasum.h:462:18: error: always_inline function '_mm256_fmadd_pd' requires target feature 'fma', but would be inlined into function 'ComputeDotAndNormSqrdsfp16' that is compiled without support for 'fma'
      anormVec = _mm256_fmadd_pd(aTop, aTop, anormVec);
                 ^
horovod/common/ops/adasum/adasum.h:463:18: error: always_inline function '_mm256_fmadd_pd' requires target feature 'fma', but would be inlined into function 'ComputeDotAndNormSqrdsfp16' that is compiled without support for 'fma'
      bnormVec = _mm256_fmadd_pd(bBot, bBot, bnormVec);
                 ^
horovod/common/ops/adasum/adasum.h:464:18: error: always_inline function '_mm256_fmadd_pd' requires target feature 'fma', but would be inlined into function 'ComputeDotAndNormSqrdsfp16' that is compiled without support for 'fma'
      bnormVec = _mm256_fmadd_pd(bTop, bTop, bnormVec);
                 ^
horovod/common/ops/adasum/adasum.h:482:24: error: always_inline function '_mm256_fmadd_ps' requires target feature 'fma', but would be inlined into function 'ScaledAddfp16' that is compiled without support for 'fma'
      MmStorePh(&a[i], _mm256_fmadd_ps(bcoeffVec, bVec, aVec));
                       ^
horovod/common/ops/adasum/adasum.h:488:31: error: always_inline function '_mm256_fmadd_ps' requires target feature 'fma', but would be inlined into function 'ScaledAddfp16' that is compiled without support for 'fma'
      MmStorePhPartial(&a[i], _mm256_fmadd_ps(bcoeffVec, bVec, aVec), len - i);
                              ^
1 warning and 14 errors generated.
INFO: Unable to build TensorFlow plugin, will skip it.
```
Full build log and environment:

* [spack-build-out.txt](https://github.com/horovod/horovod/files/4395288/spack-build-out.txt)
* [spack-build-env.txt](https://github.com/horovod/horovod/files/4395291/spack-build-env.txt)

--- a/horovod/common/ops/adasum/adasum.h
+++ b/horovod/common/ops/adasum/adasum.h
@@ -19,7 +19,7 @@
 #include <cstring>
 #include <float.h>
 
-#if __AVX__ && __F16C__
+#if __AVX__ && __F16C__ && __FMA__
 #include <emmintrin.h>
 #include <immintrin.h>
 #endif
@@ -104,7 +104,7 @@ protected:
                                               int count, double& dotProduct,
                                               double& anormsq, double& bnormsq,
                                               int layerid) {
-#if __AVX__ && __F16C__
+#if __AVX__ && __F16C__ && __FMA__
     if (horovod_datatype == DataType::HOROVOD_FLOAT16) {
       ComputeDotAndNormSqrdsfp16((uint16_t*)a, (uint16_t*)b, count, dotProduct,
                                  anormsq, bnormsq, layerid);
@@ -125,7 +125,7 @@ protected:
                                  double acoeff, void* __restrict__ a,
                                  double bcoeff, void* __restrict__ b,
                                  int layerid) {
-#if __AVX__ && __F16C__
+#if __AVX__ && __F16C__ && __FMA__
     if (horovod_datatype == DataType::HOROVOD_FLOAT16) {
       ScaledAddfp16(count, acoeff, (uint16_t*)a, bcoeff, (uint16_t*)b, layerid);
     } else
@@ -425,7 +425,7 @@ private:
   }
 
 
-#if __AVX__ && __F16C__
+#if __AVX__ && __F16C__ && __FMA__
   inline void ComputeDotAndNormSqrdsfp16(const uint16_t* __restrict__ a,
                                          const uint16_t* __restrict__ b,
                                          int len, double& dotProduct,

